low-level/files/opt/seagate/sspl/setup/resource_map/storage.py

```diff
@@ -81,8 +82,15 @@ def get_health_info(self, rpath):
         nodes = rpath.strip().split(">")
         leaf_node, _ = self.get_node_details(nodes[-1])
         if leaf_node == "storage":
-            for fru in self.storage_frus:
-                info.update({fru: self.storage_frus[fru]()})
+            # TODO should we handle >hw>* case?
+            if nodes[-1] == "controllers":
+                info.update(
+                    {"controllers": self.storage_frus["controllers"]()}
+                )
+            elif nodes[-1] == "platform_sensors":
+                info.update(
+                    {"platform_sensors": self.storage_frus["sensor-status"]()}
+                )
             info["last_updated"] = int(time.time())
```

Comment:
leaf_node is "storage" already.
so why do we need to check  `nodes[-1] == "controllers"`?
If anything else in leaf_node only, we should process. Please check else part here.

Reply:
Agree, dont think we need this if-else case, current implementation is optimal

Reply:
Understood.